### ISSUE TITLE
[sdl2:x86-window] build failure

### ISSUE THREAD

**Issue by mgabris** [microsoft/vcpkg#14298]
**Host Environment**
- OS: windows 10 
- Compiler: visual studio 2019 community

**To Reproduce**
`./vcpkg install sdl2`

**Failure logs** 
```
> .\vcpkg.exe install sdl2
...
... doing stuff ...
...
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:142 (message):
    Command failed: C:/Users/mg/srcs/vcpkg/downloads/tools/cmake-3.18.4-windows/cmake-3.18.4-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v -j17
    Working Directory: C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/x86-windows-dbg
    See logs for more information:
      C:\Users\mg\srcs\vcpkg\buildtrees\sdl2\install-x86-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:93 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/sdl2/portfile.cmake:39 (vcpkg_install_cmake)
  scripts/ports.cmake:135 (include)


Error: Building package sdl2:x86-windows failed with: BUILD_FAILED
```
- content of `C:\Users\mg\srcs\vcpkg\buildtrees\sdl2\install-x86-windows-dbg-err.log`
```
ninja: error: 'C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/src/SDL2-2-ac07e8117e.clean/src/main/windows/SDL_windows_main.c', needed by 'CMakeFiles/SDL2main.dir/src/main/windows/SDL_windows_main.c.obj', missing and no known rule to make it
```

**Additional context**
- apparently there is something wrong with sdl2 cmake rules
- or do I have wrong version of cmake / ninja? I did not install it manually, vcpkg did everything automatically, cmake 3.18.4-win32 , ninja 1.10.1-windows

**Comment by JackBoosY** [comment]
Could you provide me the configure full log and the build full log?

Thanks.

**Comment by mgabris** [comment]
Yes:
- config-x86-windows-out.log
```
[1/2] cmd /c "cd .. && "C:/Users/mg/srcs/vcpkg/downloads/tools/cmake-3.18.4-windows/cmake-3.18.4-win32-x86/bin/cmake.exe" "C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/src/SDL2-2-ac07e8117e.clean" "-DVIDEO_VULKAN=OFF" "-DSDL_STATIC=0" "-DSDL_SHARED=1" "-DFORCE_STATIC_VCRT=0" "-DLIBC=ON" "-DCMAKE_MAKE_PROGRAM=C:/Users/mg/srcs/vcpkg/downloads/tools/ninja/1.10.1-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/mg/srcs/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/mg/srcs/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Users/mg/srcs/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/Users/mg/srcs/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/Users/mg/srcs/vcpkg/packages/sdl2_x86-windows""
-- The C compiler identification is MSVC 19.27.29112.0
-- The CXX compiler identification is MSVC 19.27.29112.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe - works
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe - works
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Performing Test HAVE_WIN32_CC
-- Performing Test HAVE_WIN32_CC - Success
-- Performing Test HAVE_XINPUT_H
-- Performing Test HAVE_XINPUT_H - Success
-- Performing Test HAVE_XINPUT_GAMEPAD_EX
-- Performing Test HAVE_XINPUT_GAMEPAD_EX - Failed
-- Performing Test HAVE_XINPUT_STATE_EX
-- Performing Test HAVE_XINPUT_STATE_EX - Failed
-- Looking for d3d9.h
-- Looking for d3d9.h - found
-- Looking for d3d11_1.h
-- Looking for d3d11_1.h - found
-- Looking for ddraw.h
-- Looking for ddraw.h - found
-- Looking for dsound.h
-- Looking for dsound.h - found
-- Looking for dinput.h
-- Looking for dinput.h - found
-- Looking for dxgi.h
-- Looking for dxgi.h - found
-- Looking for mmdeviceapi.h
-- Looking for mmdeviceapi.h - found
-- Looking for audioclient.h
-- Looking for audioclient.h - found
-- 
-- SDL2 was configured with the following options:
-- 
-- Platform: Windows-10.0.19041
-- 64-bit:   FALSE
-- Compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe
-- 
-- Subsystems:
--   Atomic:	ON
--   Audio:	ON
--   Video:	ON
--   Render:	ON
--   Events:	ON
--   Joystick:	ON
--   Haptic:	ON
--   Power:	ON
--   Threads:	ON
--   Timers:	ON
--   File:	ON
--   Loadso:	ON
--   CPUinfo:	ON
--   Filesystem:	ON
--   Dlopen:	ON
--   Sensor:	ON
-- 
-- Options:
--   3DNOW                  (Wanted: ON): ON
--   ALSA                   (Wanted: OFF): OFF
--   ALSA_SHARED            (Wanted: OFF): OFF
--   ALTIVEC                (Wanted: ON): OFF
--   ARMNEON                (Wanted: ON): OFF
--   ARMSIMD                (Wanted: ON): OFF
--   ARTS                   (Wanted: OFF): OFF
--   ARTS_SHARED            (Wanted: OFF): OFF
--   ASSEMBLY               (Wanted: ON): OFF
--   ASSERTIONS             (Wanted: auto): auto
--   BACKGROUNDING_SIGNAL   (Wanted: OFF): OFF
--   CLOCK_GETTIME          (Wanted: OFF): OFF
--   DIRECTFB_SHARED        (Wanted: OFF): OFF
--   DIRECTX                (Wanted: ON): ON
--   DISKAUDIO              (Wanted: ON): ON
--   DUMMYAUDIO             (Wanted: ON): ON
--   ESD                    (Wanted: OFF): OFF
--   ESD_SHARED             (Wanted: OFF): OFF
--   FOREGROUNDING_SIGNAL   (Wanted: OFF): OFF
--   FUSIONSOUND            (Wanted: OFF): OFF
--   FUSIONSOUND_SHARED     (Wanted: OFF): OFF
--   GCC_ATOMICS            (Wanted: OFF): OFF
--   HIDAPI                 (Wanted: ON): ON
--   INPUT_TSLIB            (Wanted: OFF): OFF
--   JACK                   (Wanted: OFF): OFF
--   JACK_SHARED            (Wanted: OFF): OFF
--   KMSDRM_SHARED          (Wanted: OFF): OFF
--   LIBC                   (Wanted: ON): ON
--   LIBSAMPLERATE          (Wanted: OFF): OFF
--   LIBSAMPLERATE_SHARED   (Wanted: OFF): OFF
--   MMX                    (Wanted: ON): ON
--   NAS                    (Wanted: OFF): OFF
--   NAS_SHARED             (Wanted: OFF): OFF
--   OSS                    (Wanted: OFF): OFF
--   PTHREADS               (Wanted: OFF): OFF
--   PTHREADS_SEM           (Wanted: OFF): OFF
--   PULSEAUDIO             (Wanted: OFF): OFF
--   PULSEAUDIO_SHARED      (Wanted: OFF): OFF
--   RENDER_D3D             (Wanted: ON): ON
--   RENDER_METAL           (Wanted: OFF): OFF
--   RPATH                  (Wanted: OFF): OFF
--   SDL_DLOPEN             (Wanted: ON): OFF
--   SDL_STATIC_PIC         (Wanted: OFF): OFF
--   SDL_TEST               (Wanted: OFF): OFF
--   SNDIO                  (Wanted: OFF): OFF
--   SSE                    (Wanted: ON): ON
--   SSE2                   (Wanted: OFF): ON
--   SSE3                   (Wanted: OFF): ON
--   SSEMATH                (Wanted: OFF): OFF
--   VIDEO_COCOA            (Wanted: OFF): OFF
--   VIDEO_DIRECTFB         (Wanted: OFF): OFF
--   VIDEO_DUMMY            (Wanted: ON): ON
--   VIDEO_KMSDRM           (Wanted: OFF): OFF
--   VIDEO_METAL            (Wanted: OFF): OFF
--   VIDEO_OFFSCREEN        (Wanted: OFF): OFF
--   VIDEO_OPENGL           (Wanted: ON): ON
--   VIDEO_OPENGLES         (Wanted: ON): ON
--   VIDEO_RPI              (Wanted: OFF): OFF
--   VIDEO_VIVANTE          (Wanted: OFF): OFF
--   VIDEO_VULKAN           (Wanted: OFF): OFF
--   VIDEO_WAYLAND          (Wanted: OFF): OFF
--   VIDEO_WAYLAND_QT_TOUCH (Wanted: OFF): OFF
--   VIDEO_X11              (Wanted: OFF): OFF
--   VIDEO_X11_XCURSOR      (Wanted: OFF): OFF
--   VIDEO_X11_XINERAMA     (Wanted: OFF): OFF
--   VIDEO_X11_XINPUT       (Wanted: OFF): OFF
--   VIDEO_X11_XRANDR       (Wanted: OFF): OFF
--   VIDEO_X11_XSCRNSAVER   (Wanted: OFF): OFF
--   VIDEO_X11_XSHAPE       (Wanted: OFF): OFF
--   VIDEO_X11_XVM          (Wanted: OFF): OFF
--   WASAPI                 (Wanted: ON): OFF
--   WAYLAND_SHARED         (Wanted: OFF): OFF
--   X11_SHARED             (Wanted: OFF): OFF
-- 
--  CFLAGS:         /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP   "-IC:/Users/mg/srcs/vcpkg/buildtrees/sdl2/src/SDL2-2-ac07e8117e.clean/src/hidapi/hidapi"
--  EXTRA_CFLAGS:  
--  EXTRA_LDFLAGS: 
--  EXTRA_LIBS:    user32;gdi32;winmm;imm32;ole32;oleaut32;version;uuid;advapi32;setupapi;shell32;dinput8
-- 
--  Build Shared Library: 1
--  Build Static Library: 0
-- 
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/x86-windows-rel
[2/2] cmd /c "cd ../../x86-windows-dbg && "C:/Users/mg/srcs/vcpkg/downloads/tools/cmake-3.18.4-windows/cmake-3.18.4-win32-x86/bin/cmake.exe" "C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/src/SDL2-2-ac07e8117e.clean" "-DVIDEO_VULKAN=OFF" "-DSDL_STATIC=0" "-DSDL_SHARED=1" "-DFORCE_STATIC_VCRT=0" "-DLIBC=ON" "-DCMAKE_MAKE_PROGRAM=C:/Users/mg/srcs/vcpkg/downloads/tools/ninja/1.10.1-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/Users/mg/srcs/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/mg/srcs/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/Users/mg/srcs/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/Users/mg/srcs/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/Users/mg/srcs/vcpkg/packages/sdl2_x86-windows/debug""
-- The C compiler identification is MSVC 19.27.29112.0
-- The CXX compiler identification is MSVC 19.27.29112.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe - works
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe - works
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Performing Test HAVE_WIN32_CC
-- Performing Test HAVE_WIN32_CC - Success
-- Performing Test HAVE_XINPUT_H
-- Performing Test HAVE_XINPUT_H - Success
-- Performing Test HAVE_XINPUT_GAMEPAD_EX
-- Performing Test HAVE_XINPUT_GAMEPAD_EX - Failed
-- Performing Test HAVE_XINPUT_STATE_EX
-- Performing Test HAVE_XINPUT_STATE_EX - Failed
-- Looking for d3d9.h
-- Looking for d3d9.h - found
-- Looking for d3d11_1.h
-- Looking for d3d11_1.h - found
-- Looking for ddraw.h
-- Looking for ddraw.h - found
-- Looking for dsound.h
-- Looking for dsound.h - found
-- Looking for dinput.h
-- Looking for dinput.h - found
-- Looking for dxgi.h
-- Looking for dxgi.h - found
-- Looking for mmdeviceapi.h
-- Looking for mmdeviceapi.h - found
-- Looking for audioclient.h
-- Looking for audioclient.h - found
-- 
-- SDL2 was configured with the following options:
-- 
-- Platform: Windows-10.0.19041
-- 64-bit:   FALSE
-- Compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x86/cl.exe
-- 
-- Subsystems:
--   Atomic:	ON
--   Audio:	ON
--   Video:	ON
--   Render:	ON
--   Events:	ON
--   Joystick:	ON
--   Haptic:	ON
--   Power:	ON
--   Threads:	ON
--   Timers:	ON
--   File:	ON
--   Loadso:	ON
--   CPUinfo:	ON
--   Filesystem:	ON
--   Dlopen:	ON
--   Sensor:	ON
-- 
-- Options:
--   3DNOW                  (Wanted: ON): ON
--   ALSA                   (Wanted: OFF): OFF
--   ALSA_SHARED            (Wanted: OFF): OFF
--   ALTIVEC                (Wanted: ON): OFF
--   ARMNEON                (Wanted: ON): OFF
--   ARMSIMD                (Wanted: ON): OFF
--   ARTS                   (Wanted: OFF): OFF
--   ARTS_SHARED            (Wanted: OFF): OFF
--   ASSEMBLY               (Wanted: ON): OFF
--   ASSERTIONS             (Wanted: auto): auto
--   BACKGROUNDING_SIGNAL   (Wanted: OFF): OFF
--   CLOCK_GETTIME          (Wanted: OFF): OFF
--   DIRECTFB_SHARED        (Wanted: OFF): OFF
--   DIRECTX                (Wanted: ON): ON
--   DISKAUDIO              (Wanted: ON): ON
--   DUMMYAUDIO             (Wanted: ON): ON
--   ESD                    (Wanted: OFF): OFF
--   ESD_SHARED             (Wanted: OFF): OFF
--   FOREGROUNDING_SIGNAL   (Wanted: OFF): OFF
--   FUSIONSOUND            (Wanted: OFF): OFF
--   FUSIONSOUND_SHARED     (Wanted: OFF): OFF
--   GCC_ATOMICS            (Wanted: OFF): OFF
--   HIDAPI                 (Wanted: ON): ON
--   INPUT_TSLIB            (Wanted: OFF): OFF
--   JACK                   (Wanted: OFF): OFF
--   JACK_SHARED            (Wanted: OFF): OFF
--   KMSDRM_SHARED          (Wanted: OFF): OFF
--   LIBC                   (Wanted: ON): ON
--   LIBSAMPLERATE          (Wanted: OFF): OFF
--   LIBSAMPLERATE_SHARED   (Wanted: OFF): OFF
--   MMX                    (Wanted: ON): ON
--   NAS                    (Wanted: OFF): OFF
--   NAS_SHARED             (Wanted: OFF): OFF
--   OSS                    (Wanted: OFF): OFF
--   PTHREADS               (Wanted: OFF): OFF
--   PTHREADS_SEM           (Wanted: OFF): OFF
--   PULSEAUDIO             (Wanted: OFF): OFF
--   PULSEAUDIO_SHARED      (Wanted: OFF): OFF
--   RENDER_D3D             (Wanted: ON): ON
--   RENDER_METAL           (Wanted: OFF): OFF
--   RPATH                  (Wanted: OFF): OFF
--   SDL_DLOPEN             (Wanted: ON): OFF
--   SDL_STATIC_PIC         (Wanted: OFF): OFF
--   SDL_TEST               (Wanted: OFF): OFF
--   SNDIO                  (Wanted: OFF): OFF
--   SSE                    (Wanted: ON): ON
--   SSE2                   (Wanted: OFF): ON
--   SSE3                   (Wanted: OFF): ON
--   SSEMATH                (Wanted: OFF): OFF
--   VIDEO_COCOA            (Wanted: OFF): OFF
--   VIDEO_DIRECTFB         (Wanted: OFF): OFF
--   VIDEO_DUMMY            (Wanted: ON): ON
--   VIDEO_KMSDRM           (Wanted: OFF): OFF
--   VIDEO_METAL            (Wanted: OFF): OFF
--   VIDEO_OFFSCREEN        (Wanted: OFF): OFF
--   VIDEO_OPENGL           (Wanted: ON): ON
--   VIDEO_OPENGLES         (Wanted: ON): ON
--   VIDEO_RPI              (Wanted: OFF): OFF
--   VIDEO_VIVANTE          (Wanted: OFF): OFF
--   VIDEO_VULKAN           (Wanted: OFF): OFF
--   VIDEO_WAYLAND          (Wanted: OFF): OFF
--   VIDEO_WAYLAND_QT_TOUCH (Wanted: OFF): OFF
--   VIDEO_X11              (Wanted: OFF): OFF
--   VIDEO_X11_XCURSOR      (Wanted: OFF): OFF
--   VIDEO_X11_XINERAMA     (Wanted: OFF): OFF
--   VIDEO_X11_XINPUT       (Wanted: OFF): OFF
--   VIDEO_X11_XRANDR       (Wanted: OFF): OFF
--   VIDEO_X11_XSCRNSAVER   (Wanted: OFF): OFF
--   VIDEO_X11_XSHAPE       (Wanted: OFF): OFF
--   VIDEO_X11_XVM          (Wanted: OFF): OFF
--   WASAPI                 (Wanted: ON): OFF
--   WAYLAND_SHARED         (Wanted: OFF): OFF
--   X11_SHARED             (Wanted: OFF): OFF
-- 
--  CFLAGS:         /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /MP   "-IC:/Users/mg/srcs/vcpkg/buildtrees/sdl2/src/SDL2-2-ac07e8117e.clean/src/hidapi/hidapi"
--  EXTRA_CFLAGS:  
--  EXTRA_LDFLAGS: 
--  EXTRA_LIBS:    user32;gdi32;winmm;imm32;ole32;oleaut32;version;uuid;advapi32;setupapi;shell32;dinput8
-- 
--  Build Shared Library: 1
--  Build Static Library: 0
-- 
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/x86-windows-dbg
```
- config-x86-windows-out.err - empty
- stdout-x86-windows.log
```
-- Using cached C:/Users/mg/srcs/vcpkg/downloads/SDL2-2.0.12.tar.gz
-- Cleaning sources at C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/src/SDL2-2-ac07e8117e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/mg/srcs/vcpkg/downloads/SDL2-2.0.12.tar.gz
-- Applying patch export-symbols-only-in-shared-build.patch
-- Applying patch enable-winrt-cmake.patch
-- Applying patch disable-hidapi-for-uwp.patch
-- Applying patch fix-space-in-path.patch
-- Applying patch disable-wcslcpy-and-wcslcat-for-windows.patch
-- Applying patch fix-EventToken-header-reference.patch
-- Applying patch 0006-sdl2-Enable-creation-of-pkg-cfg-file-on-windows.patch
-- Using source at C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/src/SDL2-2-ac07e8117e.clean
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:142 (message):
    Command failed: C:/Users/mg/srcs/vcpkg/downloads/tools/cmake-3.18.4-windows/cmake-3.18.4-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v -j17
    Working Directory: C:/Users/mg/srcs/vcpkg/buildtrees/sdl2/x86-windows-dbg
    See logs for more information:
      C:\Users\mg\srcs\vcpkg\buildtrees\sdl2\install-x86-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:93 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/sdl2/portfile.cmake:39 (vcpkg_install_cmake)
  scripts/ports.cmake:135 (include)
```
- x86-windows.vcpkg_abi_info.txt
```
0006-sdl2-Enable-creation-of-pkg-cfg-file-on-windows.patch 1ba145fb3419c6692bb7cf5db66ce4112a09a473
cmake 3.18.4
disable-hidapi-for-uwp.patch 45cb21df78fad1c5f663256c0f02eedfed71b864
disable-wcslcpy-and-wcslcat-for-windows.patch 4963b8af4aca70fd58a467a67e4be0b6d431c060
enable-winrt-cmake.patch b8782ed6012e2f5d8a121623b8d5c7a25dc09d5f
export-symbols-only-in-shared-build.patch 9eb7c558bf8bb3df52be2f8ed7d75654f5806e6c
features core
fix-EventToken-header-reference.patch ebf4f15a7e2eefb2492d99516c9457b0ccfe9472
fix-space-in-path.patch 75c5082a8431917fafb40c906c1ae214af32c228
portfile.cmake 9e59d0560031592ffbb8a2fc7f5351f137cf1a62
post_build_checks 2
powershell 7.0.3
triplet e8dc96fd479b53bb0e6748625827c8268017e35a-a1c0eabb0c5177b6a8fb97a58ae398880c47b352-d8c1982fc772c9b47c816273e96145953dc4d7ea
vcpkg-cmake-wrapper.cmake c83f8cc683a1c5af100fc0debe531be2f09e0766
vcpkg.json 4fc4808786b73b4bad63c5e733ac4ddf8d55b002
vcpkg_check_features 5cc6f6260e441b945a862c1a4f275ae3824afd3c
vcpkg_configure_cmake ae0a97eff7f218b68bc4ab1b4bd661107893803e
vcpkg_copy_pdbs dbca2886a2490c948893c8250a2b40a3006cf7a9
vcpkg_download_distfile 6bdeb2a32b108188dc54b8cb85d89e361a27da65
vcpkg_extract_source_archive 40d93d67b03dd376ba314638bcfbd0f9e6e5a930
vcpkg_extract_source_archive_ex 7d3913992983ebe586e798af665c2cb68559c71a
vcpkg_fixup_cmake_targets 799f0da59bbd2e387502be06a30e3d43f2e89b49
vcpkg_fixup_pkgconfig 0ea91a7071a64da700caec415195482a4c9e775d
vcpkg_install_cmake 75d1079d6f563d87d08c8c5f63b359377aa19a6e
vcpkg_replace_string 99deefc316db487a541ba0b25d6249a0b86e2186
```
- multiple `patch-xxx` logs - all successfull, there were 7 of those, if needed I can paste them also here
- extract-err, extract-out were empty
- install-out empty, install-err is in previous comment

**Comment by mgabris** [comment]
and btw, I just noticed vcpkg downloaded 32bit version of sdl2, is it ok? I have (as I think almost everyone now) 64bit windows, sdl does have 64bit version of their lib

**Comment by StarGate-One** [comment]
If you want to create 64 bit windows binaries, use one of the following from the [vcpkg-root] folder/directory:
1. a. open cmd prompt
    b.`set VCPKG_DEFAULT_TRIPLET=x64-windows`  -or- `set VCPKG_DEFAULT_TRIPLET=x64-windows-static`  
    c. `vcpkg install sdl2`
2. a. open powershel prompt
   b. `$ENV:VCPKG_DEFAULT_TRIPLET = "x64-windows"` -or- `$ENV:VCPKG_DEFAULT_TRIPLET="x64-windows-static`
   c. `vcpkg install sdl2`
3. ` vcpkg install sdl2:x64-windows` -or- `vcpkg install sdl2:x64-windows-static`
4. `vcpkg install sdl2 --triplet=x64-windows` -or- `vcpkg install sdl2 --triplet=x64-windows-static`

Note: For windows (x64 or x86) the default binary build is x86 (32-bit).

**Comment by JackBoosY** [comment]
In order to ensure that there will be no permission problems, please clone vcpkg in a non-system disk, and try again.

Thanks.

**Comment by mgabris** [comment]
Thanks for suggestion, running vcpkg on different drive did solve that issue, `sdl2` built without errors there. 

This was quite weird to me, I was only user on this pc, I did not mess with permissions, run vcpkg/git/powershell as admin... 

So I played with it a bit and it seems WSL turns on case sensitivity in NTFS if you have drive mounted also inside WSL. This seems to mess with vcpk and cmake inside it. When I toggled it off on vcpkg folder (with `(Get-ChildItem -Recurse -Directory).FullName | ForEach-Object {fsutil.exe file setCaseSensitiveInfo $_ disable}`), it started working also on main drive.

Anyway thanks for help, issue was on my side :(
